cfe_internal/enterprise/federation/federation.cf

```diff
@@ -133,18 +133,51 @@ bundle agent federation_manage_files
         perms => default:mog( "600", "root", "root" );
 
     # TODO: Render these from templates:
-    am_feeder::
-      "$(cfengine_enterprise_federation:config.bin_dir)/dump.sh"
+
+    am_feeder|am_transporter|am_superhub::
+      # TODO: Instrument augments
+      "$(cfengine_enterprise_federation:config.bin_dir)/config.sh"
         create => "true",
+        template_method => "mustache",
+        edit_template => "$(this.promise_dirname)/../../../templates/federated_reporting/config.sh.mustache",
         perms => default:mog( "600", "root", "root" );
-    am_transporter::
-      "$(cfengine_enterprise_federation:config.bin_dir)/transport.sh"
+
+      # TODO: Instrument augments
+      "$(cfengine_enterprise_federation:config.bin_dir)/log.sh"
         create => "true",
+        template_method => "mustache",
+        edit_template => "$(this.promise_dirname)/../../../templates/federated_reporting/log.sh.mustache",
+        template_data => '{ "logger": "log", "logger_args": "", "timestamp_cmd": "date --iso-8601=seconds"}',
+        perms => default:mog( "600", "root", "root" );
+
+      "$(cfengine_enterprise_federation:config.bin_dir)/parallel.sh"
+        copy_from => local_dcp( "$(this.promise_dirname)/../../../templates/federated_reporting/parallel.sh" ),
         perms => default:mog( "600", "root", "root" );
+
+    am_feeder::
+      "$(cfengine_enterprise_federation:config.bin_dir)/dump.sh"
+        copy_from => local_dcp( "$(this.promise_dirname)/../../../templates/dump.sh" ),
+        perms => default:mog( "700", "root", "root" );
+
+    am_transporter::
+      "$(cfengine_enterprise_federation:config.bin_dir)/transport.sh"
+        copy_from => local_dcp( "$(this.promise_dirname)/../../../templates/transport.sh" ),
+        perms => default:mog( "700", "root", "root" );
+
     am_superhub::
       "$(cfengine_enterprise_federation:config.bin_dir)/import.sh"
-        create => "true",
+        copy_from => local_dcp( "$(this.promise_dirname)/../../../templates/import.sh" ),
+        perms => default:mog( "700", "root", "root" );
+
+      "$(cfengine_enterprise_federation:config.bin_dir)/import_file.sh"
+        copy_from => local_dcp( "$(this.promise_dirname)/../../../templates/import_file.sh" ),
+        perms => default:mog( "700", "root", "root" );
+
+      "$(cfengine_enterprise_federation:config.bin_dir)/filter.sed"
+        copy_from => local_dcp( "$(this.promise_dirname)/../../../templates/filter.sed" ),
         perms => default:mog( "600", "root", "root" );
+
```

Comment:
@nickanderson @vpodzime This doesn't seem quite right. The files copied over by `cf-agent -KIf update.cf` don't include `*.sed` I'm pretty sure. Reading update_policy.cf and update_def.cf which has "input_name_patterns" which doesn't include `*.sed`.
Also the config.sh and import_file.sh scripts seem to expect multiple sed filter files in /opt/cfengine/federation/superhub/import/filters directory but I don't see anything that would copy `*.sed` to that location in the policy.

Reply:
You're right. add to input name patterns or rename to .txt and copy to .sed?

Reply:
I am working on a fix for the import/filters path creation and copying *.sed over to there.

Reply:
I don't think *.sed makes sense for i2 so think just one filter.sed is fine for now. Easier in policy anyway.

Reply:
Yes. Also, if a user wants to install sed files, they can add their own promises in their own bundles and files.